app/src/androidTest/java/org/simple/clinic/encounter/EncounterRepositoryAndroidTest.kt

```diff
@@ -184,24 +194,65 @@ class EncounterRepositoryAndroidTest {
+    repository.saveBloodPressureMeasurement(alreadySavedBloodPressure).blockingAwait()
+    repository.saveBloodPressureMeasurement(bloodPressureToBeSaved).blockingAwait()
 
     //then
-    val savedBp = appDatabase.bloodPressureDao().bloodPressure(bpUuid).blockingFirst()
-    val encounters = appDatabase.encountersDao().recordsWithSyncStatus(SyncStatus.PENDING).blockingFirst()
+    val updatedBp = appDatabase.bloodPressureDao().bloodPressure(bpUuid).blockingFirst()
+    val encounters = appDatabase.encountersDao().recordsWithSyncStatus(PENDING).blockingFirst()
 
     assertThat(encounters.size).isEqualTo(1)
 
     with(encounters.first()) {
       with(encounter) {
         assertThat(encounteredOn).isEqualTo(encounteredDate)
-        assertThat(createdAt).isEqualTo(savedBp.createdAt)
-        assertThat(updatedAt).isEqualTo(savedBp.updatedAt)
-        assertThat(deletedAt).isEqualTo(savedBp.deletedAt)
-        assertThat(syncStatus).isEqualTo(savedBp.syncStatus)
+        assertThat(createdAt).isEqualTo(updatedBp.createdAt)
+        assertThat(updatedAt).isEqualTo(updatedBp.updatedAt)
+        assertThat(deletedAt).isEqualTo(updatedBp.deletedAt)
+        assertThat(syncStatus).isEqualTo(updatedBp.syncStatus)
       }
       assertThat(bloodPressures).isEqualTo(listOf(alreadySavedBloodPressure, bloodPressureToBeSaved))
     }
   }
+
+  @Test
+  fun when_a_bp_is_updated_then_its_encounter_should_also_get_updated() {
+    //given
+    val patientUuid = UUID.fromString("6f725ffd-7008-4018-9e7c-33346964a0c1")
+    val facilityUuid = UUID.fromString("22bdeedb-061e-4a17-8739-e946a4206593")
+    val bpUuid = UUID.fromString("2edd4c06-e2de-4a18-a8e7-43e2e30c9aba")
+    val encounteredDate = LocalDate.parse("2018-01-01")
+    val bloodPressureMeasurement = testData.bloodPressureMeasurement(
+        uuid = bpUuid,
+        patientUuid = patientUuid,
+        facilityUuid = facilityUuid,
+        recordedAt = Instant.parse("2018-01-01T00:00:00Z"),
+        createdAt = Instant.parse("2018-01-01T00:00:00Z"),
+        updatedAt = Instant.parse("2018-01-01T00:00:00Z"),
+        syncStatus = DONE,
+        encounterUuid = generateEncounterUuid(facilityUuid, patientUuid, encounteredDate)
+    )
+    repository.saveBloodPressureMeasurement(bloodPressureMeasurement).blockingAwait()
+
+    //when
+    val updatedMeasurement = bloodPressureMeasurement.copy(systolic = 200)
+    repository.updateBloodPressure(updatedMeasurement).blockingAwait()
+
+    //then
+    val updatedBp = appDatabase.bloodPressureDao().bloodPressure(bpUuid).blockingFirst()
+    val encounters = appDatabase.encountersDao().recordsWithSyncStatus(PENDING).blockingFirst()
+
+    assertThat(encounters.size).isEqualTo(1)
+    with(encounters.first()) {
+      assertThat(bloodPressures.first()).isEqualTo(updatedMeasurement)
+
+      with(encounter) {
+        assertThat(encounteredOn).isEqualTo(encounteredDate)
+        assertThat(createdAt).isEqualTo(updatedBp.createdAt)
```

Comment:
Curious, why is the encounter's `createdAt` and `deletedAt` properties are being updated?

1. Shouldn't we not update the `createdAt` property?
2. If only all the BP entries are deleted, then we update the `deletedAt` property for an `Encounter`?

Am I missing something here?

Reply:
Good catch!
Your understanding is correct. I added these assertions to ensure that only `updatedAt` is changed.
I will change these assertions.